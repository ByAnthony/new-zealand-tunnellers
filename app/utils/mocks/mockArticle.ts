import { AboutUsArticle } from "../../types/article";

export const mockArticle: AboutUsArticle = {
  id: "about-us",
  title: "Our Story",
  section: [
    { title: "Title 1", text: "This is the first paragraph of our story." },
    { title: "Title 2", text: "This is the second paragraph of our story." },
  ],
  image: [
    {
      file: "our-story.jpg",
      title: null,
      photographer: null,
      reference: null,
      alt: "A picture of our team",
    },
  ],
};
