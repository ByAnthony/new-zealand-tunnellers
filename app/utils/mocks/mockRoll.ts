import { Tunneller } from '@/app/types/roll';
import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails: Tunneller = {
  id: mockId,
  name: johnDoe,
  birthDate: '1886',
  deathDate: '1952',
};

export const mockDetailsList: Tunneller[] = [
  mockDetails,
  {
    ...mockDetails,
    id: 1,
    name: alexanderDriscott,
  },
];

export const mockRoll: Record<string, Tunneller[]> = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
