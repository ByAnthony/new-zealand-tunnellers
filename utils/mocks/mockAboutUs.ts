import { AboutUsArticle } from "@/types/article";

import {
  mockSectionOne,
  mockSectionTwo,
  mockTitle,
  mockTopImage,
} from "./mockArticle";

export const mockAboutUs: AboutUsArticle = {
  id: "my-awesome-article",
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
