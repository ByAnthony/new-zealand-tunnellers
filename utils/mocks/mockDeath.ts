import { Death } from "@/types/tunneller";

import { mockDeath } from "./mockTunneller";

export const mockAfterWarDeath: Death = {
  ...mockDeath,
  warInjuriesDeathAfterWar: true,
};
