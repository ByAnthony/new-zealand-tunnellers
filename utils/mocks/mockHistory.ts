import { HistoryChapter } from "@/types/homepage";

export const mockHistory: HistoryChapter[] = [
  {
    id: "where-all-started",
    chapter: 1,
    title: "Where All\\Started",
    image: "where-all-started.jpg",
  },
  {
    id: "the-journey-ahead",
    chapter: 2,
    title: "The Journey\\Ahead",
    image: "the-journey-ahead.jpg",
  },
  {
    id: "discovering-the-war",
    chapter: 3,
    title: "Discovering\\The War",
    image: "discovering-the-war.jpg",
  },
  {
    id: "below-the-surface",
    chapter: 4,
    title: "Below\\The Surface",
    image: "below-the-surface.jpg",
  },
  {
    id: "back-to-the-surface",
    chapter: 5,
    title: "Back\\To The Surface",
    image: "back-to-the-surface.jpg",
  },
  {
    id: "reconversion-of-activities",
    chapter: 6,
    title: "Reconversion\\Of Activities",
    image: "reconversion-of-activities.jpg",
  },
  {
    id: "returnin-home",
    chapter: 7,
    title: "Returning\\Home",
    image: "returning-home.jpg",
  },
];
