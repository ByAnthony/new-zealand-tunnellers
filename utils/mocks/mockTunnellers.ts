import { Tunneller, TunnellerWithFullNameData } from "@/types/tunnellers";

const mockTunnellerData1: TunnellerWithFullNameData = {
  id: 1,
  forename: "John",
  surname: "Doe",
  birthYear: "1886",
  deathYear: "1952",
  fullName: "John Doe",
};

const mockTunnellerData2: TunnellerWithFullNameData = {
  id: 2,
  forename: "Biff",
  surname: "Tanen",
  birthYear: "1897",
  deathYear: null,
  fullName: "William Smith",
};

const mockTunnellerData3: TunnellerWithFullNameData = {
  id: 3,
  forename: "Emmett",
  surname: "Brown",
  birthYear: null,
  deathYear: "1935",
  fullName: "Emmett Brown",
};

const mockTunnellerData4: TunnellerWithFullNameData = {
  id: 4,
  forename: "Marty",
  surname: "McFly",
  birthYear: null,
  deathYear: null,
  fullName: "Marty McFly",
};

export const mockTunnellersData: TunnellerWithFullNameData[] = [
  mockTunnellerData1,
  mockTunnellerData2,
  mockTunnellerData3,
  mockTunnellerData4,
];

const mockTunneller1: Tunneller = {
  id: 1,
  name: { forename: "John", surname: "Doe" },
  birthYear: "1886",
  deathYear: "1952",
  fullName: "John Doe",
};

const mockTunneller2: Tunneller = {
  id: 2,
  name: { forename: "Biff", surname: "Tanen" },
  birthYear: "1897",
  deathYear: null,
  fullName: "William Smith",
};

const mockTunneller3: Tunneller = {
  id: 3,
  name: { forename: "Emmett", surname: "Brown" },
  birthYear: null,
  deathYear: "1935",
  fullName: "Emmett Brown",
};

const mockTunneller4: Tunneller = {
  id: 4,
  name: { forename: "Marty", surname: "McFly" },
  birthYear: null,
  deathYear: null,
  fullName: "Marty McFly",
};

export const mockTunnellers: Record<string, Tunneller[]> = {
  B: [mockTunneller3],
  D: [mockTunneller1],
  M: [mockTunneller4],
  S: [mockTunneller2],
};
