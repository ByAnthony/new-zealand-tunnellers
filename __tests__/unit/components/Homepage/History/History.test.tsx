import { render, screen } from "@testing-library/react";

import { History } from "@/components/HomePage/History/History";
import { mockHistory } from "@/utils/mocks/mockHistory";

describe("History", () => {
  test("matches the snapshot", () => {
    const { asFragment } = render(<History articles={mockHistory} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the component correctly", () => {
    render(<History articles={mockHistory} />);

    mockHistory.forEach((article) => {
      const articleCard = screen.getByRole("link", {
        name: new RegExp(
          `Go to Chapter ${article.chapter}: ${article.title.replace(/\\/g, " ")}`,
          "i",
        ),
      });
      expect(articleCard).toBeInTheDocument();

      const articleCardDiv = articleCard.closest("div");
      expect(articleCardDiv).toHaveStyle(
        `backgroundImage: url(../images/history/${article.image})`,
      );

      const titleLine1 = screen.getByText(splitTitle(article.title)[0]);
      const titleLine2 = screen.getByText(splitTitle(article.title)[1]);
      const titleLine3 = screen.getByText(`Chapter ${article.chapter}`);
      expect(titleLine1).toBeInTheDocument();
      expect(titleLine2).toBeInTheDocument();
      expect(titleLine3).toBeInTheDocument();
    });
  });
});

const splitTitle = (string: string) => {
  const split = string.split("\\");
  return split;
};
