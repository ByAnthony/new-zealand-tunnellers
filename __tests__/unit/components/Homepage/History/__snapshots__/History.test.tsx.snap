// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="history-chapter"
  >
    <div
      class="chapter-cards-wrapper"
      id="history"
    >
      <div
        class="chapter-cards-menu"
      >
        <h3>
          History of the Company
        </h3>
        <div
          class="chapter-cards-nav"
        >
          <button
            aria-label="Scroll to the left to see previous chapters"
            disabled=""
            style="color: rgb(44, 46, 47); border-color: rgb(44, 46, 47); background-color: rgb(24, 26, 27); cursor: not-allowed;"
            type="button"
          >
            ←
          </button>
          <button
            aria-label="Scroll to the right to see next chapters"
            type="button"
          >
            →
          </button>
        </div>
      </div>
      <div
        class="chapter-cards"
      >
        <div
          class="chapter-card"
          style="background-image: url(../images/history/where-all-started.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 1: Where All Started"
            class="link-button"
            href="/history/where-all-started"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    Where All
                  </span>
                  <span
                    class="title-line-2"
                  >
                    Started
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 1
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="chapter-card"
          style="background-image: url(../images/history/the-journey-ahead.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 2: The Journey Ahead"
            class="link-button"
            href="/history/the-journey-ahead"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    The Journey
                  </span>
                  <span
                    class="title-line-2"
                  >
                    Ahead
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 2
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="chapter-card"
          style="background-image: url(../images/history/discovering-the-war.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 3: Discovering The War"
            class="link-button"
            href="/history/discovering-the-war"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    Discovering
                  </span>
                  <span
                    class="title-line-2"
                  >
                    The War
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 3
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="chapter-card"
          style="background-image: url(../images/history/below-the-surface.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 4: Below The Surface"
            class="link-button"
            href="/history/below-the-surface"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    Below
                  </span>
                  <span
                    class="title-line-2"
                  >
                    The Surface
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 4
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="chapter-card"
          style="background-image: url(../images/history/back-to-the-surface.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 5: Back To The Surface"
            class="link-button"
            href="/history/back-to-the-surface"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    Back
                  </span>
                  <span
                    class="title-line-2"
                  >
                    To The Surface
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 5
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="chapter-card"
          style="background-image: url(../images/history/reconversion-of-activities.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 6: Reconversion Of Activities"
            class="link-button"
            href="/history/reconversion-of-activities"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    Reconversion
                  </span>
                  <span
                    class="title-line-2"
                  >
                    Of Activities
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 6
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="chapter-card"
          style="background-image: url(../images/history/returning-home.jpg); background-size: cover; background-position: center center;"
        >
          <a
            aria-label="Go to Chapter 7: Returning Home"
            class="link-button"
            href="/history/returnin-home"
          >
            <div
              class="chapter-card-dimmer"
            >
              <div
                class="chapter-card-content"
              >
                <div>
                  <span
                    class="title-line-1"
                  >
                    Returning
                  </span>
                  <span
                    class="title-line-2"
                  >
                    Home
                  </span>
                  <span
                    class="title-line-3"
                  >
                    Chapter 7
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
